### DISCUSSION TITLE
Render GeoJson Layer on iTwin viewport Map

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#4103] [General]
what is the best way to render Geojson Layer on maps in viewport?? can We do it with decorators only easily ??
@pmconne @aruniverse

**Comment by aruniverse** [comment]
Also adding @mdastous-bentley who might know more

**Reply by mdastous-bentley** [reply]
yes, I think decorators API is the only way right now.  @pmconne  might have other suggestions.

The other way would be to use the GeoConnector to import the data as a model.

**Reply by Rahul-0108** [reply]
Basically , I want to render any type of geojson feature like linestring , polygon, MultiLineString on the map ,
Also I want to render 3D map layer on the viewport like this based on the data:
![image](https://user-images.githubusercontent.com/53996840/184902989-9d0048fd-a07b-4d30-8a35-eb4fbdf1b677.png)

so view decorations is the way to do this ??
 It will be very complex for rendering the geojson layers with view decorations I think
Any suggestion on this ??

**Reply by mdastous-bentley** [reply]
Can you please clarify the requirements.  
Do you need to render the geometries client-side using iTwin.js (If so why)?  Where is going to be stored the geojson and how it's going to be published?

Does importing your geojson into the imodel would be a valid approach ?

Thanks,

Michel

**Reply by Rahul-0108** [reply]
@mdastous-bentley 
I want to show every thing in blank connection. using my backend api, I will fetch the json from my api and show it on the map.
Everything We want from frontend side only for rendering in ui.

**Reply by pmconne** [reply]
You have a variety of options.

1. Convert the GeoJson to 3d tiles and attach the tileset to your view as a context reality model. This is probably the best option if you have a large amount of geometry. [Here's a sample](https://www.itwinjs.org/sandboxes/iTwinPlatform/Reality%20Data) for attaching reality model. Web search will find various options for converting to 3d tiles.
2. Convert the GeoJson to glTF and load that as a a decoration. Web search can find you conversion options. This is an ok option if you have a relatively small amount of geometry. [Here's an example](https://www.itwinjs.org/sandboxes/NathanElrod/glTF%20Decorator) of a glTF decoration.
3. Read the GeoJson directly and use it to create decoration graphics using a GraphicBuilder. GeoJson is a pretty simple format. [Here's some code](https://github.com/iTwin/itwinjs-core/tree/master/test-apps/imodel-from-geojson) that reads it - in this example, we're converting it to a model inside an iModel but most of the logic would be the same when creating decorations instead.
